### PR TITLE
[REVIEW] Add python bindings for porter stemmer measure functionality

### PR DESCRIPTION
**Pull request by VibhuJawa**
This PR adds python bindings for porter stemmer measure functionality and is a pre requisite for closing https://github.com/rapidsai/custrings/issues/361  .

### CODE REVIEW

File: python/nvstrings/cpp/pytext.cpp [hunk 1]
```diff
@@ -663,6 +707,9 @@ static PyMethodDef s_Methods[] = {
     { "n_edit_distance", n_edit_distance, METH_VARARGS, "" },
     { "n_create_ngrams", n_create_ngrams, METH_VARARGS, "" },
     { "n_scatter_count", n_scatter_count, METH_VARARGS, "" },
+    { "n_porter_stemmer_measure", n_porter_stemmer_measure, METH_VARARGS, "" },
+
+
```
Comment (davidwendt) [inline python/nvstrings/cpp/pytext.cpp]: Looks good. Please remove these extra blank lines.

File: python/nvstrings/cpp/pytext.cpp [hunk 2]
```diff
@@ -649,6 +649,50 @@ static PyObject* n_scatter_count( PyObject* self, PyObject* args )
...
+static PyObject* n_porter_stemmer_measure( PyObject* self, PyObject* args )
+{
+
+    PyObject* pystrs = PyTuple_GetItem(args,0);
+    NVStrings* strs = strings_from_object(pystrs);
+
+     if( strs==0 )
+        Py_RETURN_NONE;
+
+    const char * vowels = "aeiou" ;
```
Comment (davidwendt) [inline python/nvstrings/cpp/pytext.cpp]: Are you sure you want to hardcode these?  Perhaps make them defaults inside the python method?

File: python/nvstrings/nvtext.py [hunk 3]
```diff
@@ -346,3 +346,27 @@ def scatter_count(strs, counts):
...
+        The strings for this operation
+    devptr : GPU memory pointer
+        Must be able to hold at least strs.size() of int32 values.
+
+    Examples
+    --------
+    >>> import nvstrings, nvtext
+    >>> s = nvstrings.to_device(["hello","goodbye",""])
+    >>> n = nvtext.porter_stemmer_measure(s)
+    >>> print(n)
```
Comment (davidwendt) [inline python/nvstrings/nvtext.py]: Show the output of the `print(n)` in here as well.

File: python/nvstrings/cpp/pytext.cpp [hunk 4]
```diff
@@ -649,6 +649,50 @@ static PyObject* n_scatter_count( PyObject* self, PyObject* args )
...
 
+//
+static PyObject* n_porter_stemmer_measure( PyObject* self, PyObject* args )
+{
+
+    PyObject* pystrs = PyTuple_GetItem(args,0);
+    NVStrings* strs = strings_from_object(pystrs);
+
+     if( strs==0 )
+        Py_RETURN_NONE;
```
Comment (davidwendt) [inline python/nvstrings/cpp/pytext.cpp]: You could check this in the python code too and throw a `ValueError` perhaps.

Comment (VibhuJawa) [reply python/nvstrings/cpp/pytext.cpp]: Added check , thanks for the review. 

https://github.com/rapidsai/cudf/blob/170e59be2442c079e56a950799c4c5e954712f33/python/nvstrings/nvtext.py#L379-L380

File: python/nvstrings/nvtext.py (hunk 3 above)
Comment (VibhuJawa) [reply python/nvstrings/nvtext.py]: Added, 

https://github.com/rapidsai/cudf/blob/170e59be2442c079e56a950799c4c5e954712f33/python/nvstrings/nvtext.py#L377

File: python/nvstrings/cpp/pytext.cpp (hunk 1 above)
Comment (VibhuJawa) [reply python/nvstrings/cpp/pytext.cpp]: Removed with commit 

https://github.com/rapidsai/cudf/pull/2777/commits/9ffa721fb37a63b03513ba84e8256ecd7f547f89

File: python/nvstrings/cpp/pytext.cpp (hunk 2 above)
Comment (VibhuJawa) [reply python/nvstrings/cpp/pytext.cpp]: Changed to default-python arguments.

 I dont think it should matter though as this is an internal functions to the algorithm. 

https://github.com/rapidsai/cudf/blob/9ffa721fb37a63b03513ba84e8256ecd7f547f89/python/nvstrings/nvtext.py#L351


https://github.com/rapidsai/cudf/blob/170e59be2442c079e56a950799c4c5e954712f33/python/nvstrings/cpp/pytext.cpp#L662-L670

**Comment by VibhuJawa** [comment]
rerun tests

**Comment by VibhuJawa** [comment]
@davidwendt , It would be amazing if you could review the changes i mage according to your suggestions so that we can get this merged in. Thanks.

**Pull request merged**